src/smart-components/access-management/UsersTable.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from 'react';
+import { useDataViewSelection, useDataViewPagination } from '@patternfly/react-data-view/dist/dynamic/Hooks';
+import { BulkSelect, BulkSelectValue } from '@patternfly/react-component-groups/dist/dynamic/BulkSelect';
+import { DataView } from '@patternfly/react-data-view/dist/dynamic/DataView';
+import { DataViewToolbar } from '@patternfly/react-data-view/dist/dynamic/DataViewToolbar';
+import { DataViewTable } from '@patternfly/react-data-view/dist/dynamic/DataViewTable';
+import { Pagination } from '@patternfly/react-core';
+import { ActionsColumn } from '@patternfly/react-table';
+import { fetchUsers } from '../../helpers/user/user-helper';
+
+const COLUMNS: string[] = ['Username', 'Email', 'First name', 'Last name', 'Status', 'Org admin'];
+
+const ROW_ACTIONS = [
+  { title: 'Add to user group', onClick: () => console.log('ADD TO USER GROUP') },
+  { title: 'Remove from user group', onClick: () => console.log('REMOVE FROM USER GROUP') },
+];
+
+const PER_PAGE_OPTIONS = [
+  { title: '5', value: 5 },
+  { title: '10', value: 10 },
+  { title: '20', value: 20 },
+  { title: '50', value: 50 },
+  { title: '100', value: 100 },
+];
+
+const OUIA_ID = 'iam-users-table';
+
+interface User {
+  username: string;
+  email: string;
+  first_name: string;
+  last_name: string;
+  is_active: boolean;
+  is_org_admin: boolean;
+}
+
+const UsersTable: React.FunctionComponent = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [totalCount, setTotalCount] = useState<number>(0);
+  const [_, setLoading] = useState<boolean>(true);
+  const [error, setError] = useState<string | null>(null);
+
+  const pagination = useDataViewPagination({ perPage: 20 });
+  const { page, perPage, onSetPage, onPerPageSelect } = pagination;
+
+  const selection = useDataViewSelection({ matchOption: (a: User, b: User) => a.username === b.username });
+  const { selected, onSelect, isSelected } = selection;
+
+  useEffect(() => {
+    const loadUsers = async () => {
+      setLoading(true);
+      try {
+        const result = await fetchUsers({
+          limit: perPage,
+          offset: (page - 1) * perPage,
+          orderBy: 'username',
+          usesMetaInURL: null,
+        });
+        setUsers(result.data);
+        setTotalCount(result.meta.count);
+      } catch (err) {
+        setError('Failed to load users');
+        console.error(err);
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    loadUsers();
+  }, [page, perPage]);
+
+  const handleBulkSelect = (value: BulkSelectValue) => {
+    if (value === BulkSelectValue.none) {
+      onSelect(false);
+    } else if (value === BulkSelectValue.page) {
+      onSelect(true, users);
+    } else if (value === BulkSelectValue.all) {
+      onSelect(true);
```

Comment:
```suggestion
    } else if (value === BulkSelectValue.nonePage) {
      onSelect(false, rows);
```

This should fix the select - the all option is not needed with the pagination